packages/date-time/src/components/Calendar/Calendar.base.tsx

```diff
@@ -81,14 +81,22 @@ const DEFAULT_PROPS: Partial<ICalendarProps> = {
 };
 
 function useDateState({ value, today = new Date(), onSelectDate }: ICalendarProps) {
+  /** The currently selected date in the calendar */
+  const [selectedDate = today, setSelectedDate] = useControllableValue(value, today);
+
   /** The currently focused date in the day picker, but not necessarily selected */
   const [navigatedDay = today, setNavigatedDay] = React.useState(value);
 
   /** The currently focused date in the month picker, but not necessarily selected */
   const [navigatedMonth = today, setNavigatedMonth] = React.useState(value);
 
-  /** The currently selected date in the calendar */
-  const [selectedDate = today, setSelectedDate] = useControllableValue(value, today);
+  /** If using a controlled value, when that value changes, navigate to that date */
+  const [lastSelectedDate = today, setLastSelectedDate] = React.useState(value);
+  if (value && lastSelectedDate.valueOf() !== value.valueOf()) {
```

Comment:
Coming in late here (sorry) but I'm pretty sure we shouldn't be setting state while rendering; seems like that could lead to unpredictable behavior. Could you do this with a ref (for the old value) and useEffect instead?